### PR TITLE
Convert to samples/counts based on available MPs

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**

In #1123 we changed the criteria for applying `measurements_from_samples` and `measurements_from_counts` to be based on whether or not there are supported `observables`. Previously the decision in preprocess had been based on supported measurement processes instead of supported observables. The reason for this change is that those transforms don't only change the measurement type - they also go a step further and diagonalize the observables to the readout basis. This is potentially a separate functionality from modifying measurement types. 

There was discussion of this being handled by a future transform that is more specific to selectively transforming unsupported MPs without changing the observables. However, that isn't implemented yet, and the QRack plugin in analytic mode only supports Counts and Samples, so its needed now.

**Description of the Change:**
This PR reintroduces the check for supported measurement types being limited to a subset of {"Counts", "Samples"} in when adding measurement transforms to the device TransformProgram.

**Benefits:**
The QRack device keeps working.

**Possible Drawbacks:**
Depending what the backend supports, arguably the TOML file loading is what should change here - if the observables are only supported in `analytic` mode, then the `qjit_capabilities` in finite-shots mode should have an empty list of `observables`, and then this change would be unnecessary. So maybe I'm not fixing this in the most logical place? But I'm not sure about that, and I'm sure this will work.

### CODE REVIEW

**Review by rmoyard — APPROVED** [review]
Thanks for the quick change!

File: frontend/catalyst/device/qjit_device.py [hunk 1]
```diff
@@ -440,6 +440,17 @@ def _measurement_transform_program(self):
             else:
                 raise RuntimeError("The device does not support observables or sample/counts")
 
+        elif self.measurement_processes in [{"Sample"}, {"Counts", "Sample"}, {"Counts"}]:
+            # ToDo: this branch should become unneccessary when selective conversion of unsupported MPs is finished, see ToDo below
+            if not split_non_commuting in measurement_program:
+                measurement_program.add_transform(split_non_commuting)
+            mp_transform = (
```
Comment (rmoyard) [inline frontend/catalyst/device/qjit_device.py:447]: @lillian542 Not sure that the tests trigger those lines?

Comment (lillian542) [reply frontend/catalyst/device/qjit_device.py:448]: The relevant test was indented and didn't run 🙈 Should be good now.

**Review by rmoyard — APPROVED** [review]
Thanks for the quick fix :+1:

**Pull request merged**